src/main/java/mateusz/michal/chat/Service/SlugService.java

```diff
@@ -0,0 +1,66 @@
+package mateusz.michal.chat.Service;
+
+import mateusz.michal.chat.Repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Random;
+
+@Service
+public class SlugService {
+
+    @Autowired
+    UserRepository userRepository;
+
+    public String generateSlugFromName(String name) {
+        StringBuilder result = new StringBuilder();
+        name = name.toLowerCase();
+        for (int i = 0; i < name.length(); ++i) {
+            switch (name.charAt(i)) {
+                case 'ą':
+                    result.append('a');
+                    break;
+                case 'ć':
+                    result.append('c');
+                    break;
+                case 'ę':
+                    result.append('e');
+                    break;
+                case 'ł':
+                    result.append('l');
+                    break;
+                case 'ń':
+                    result.append('n');
+                    break;
+                case 'ó':
+                    result.append('o');
+                    break;
+                case 'ś':
+                    result.append('s');
+                    break;
+                case 'ź':
+                    result.append('z');
+                    break;
+                case 'ż':
+                    result.append('z');
+                    break;
+                case ' ':
+                    result.append("-");
+                default:
+                    if (Character.isDigit(name.charAt(i)) || Character.isLowerCase(name.charAt(i)))
+                        result.append(name.charAt(i));
+            }
+        }
+        return result.toString();
+    }
+
+    public String generateUniqueSlug(String name){
+        String convertedSlug = generateSlugFromName(name);
+
+        while(userRepository.findBySlug(convertedSlug) != null){
+            Random random = new Random();
+            convertedSlug += random.nextInt(1000);
```

Comment:
nit: a może po prostu dodawać kolejny numer, zamiast losowy? ;)

Reply:
2. ważne - jak bot (albo użytkownicy) zrobi wszystkie 1000 kont (z końcówkami od 0 do 999), serwer zawiesi się...

Reply:
+= dokleja do końcówki Stringa. 
Rejestruje się I osoba -> dostaje unikalnego sluga
Rejestruje się II osoba z slugiem identycznym jak slug I osoby -> losowana jest liczba z przedziału 0-1000 i doklejana jest ona na końcu sluga 
Rejestruje się III osoba z slugiem identycznym jak slug I osoby i akurat wylosowana zostanie taka sama liczba -> losowana jest kolejna liczba i zostaje ona doklejona do końca sluga. W efektcie po II losowaniach mamy 1kk możliwości.

Reply:
Nie uważasz że to overkill? Tj. rozumiem że to zoptymalizuje liczbę przeszukań w bazie (w porównaniu do dodawania kolejnej liczby), ale skończymy ze slugami w stylu `mateusz12454622345123412`, kiedy mogłoby być np `mateusz145`, no ale to już zostawiam jako nit - nie musimy tego zmieniać ;)